dll/ntdll/nt_0600/ldr/ldrnotify.c

```diff
@@ -0,0 +1,148 @@
+/*
+ * PROJECT:     ReactOS NT Layer/System API
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     DLL Load Notification Implementation
+ * COPYRIGHT:   Copyright 2024 Ratin Gao <ratin@knsoft.org>
+ */
+
+#include "ntdll_vista.h"
+
+typedef struct _LDR_DLL_NOTIFICATION_ENTRY
+{
+    LIST_ENTRY List;
+    PLDR_DLL_NOTIFICATION_FUNCTION Callback;
+    PVOID Context;
+} LDR_DLL_NOTIFICATION_ENTRY, *PLDR_DLL_NOTIFICATION_ENTRY;
+_STATIC_ASSERT(FIELD_OFFSET(LDR_DLL_NOTIFICATION_ENTRY, List) == 0);
+
+static RTL_STATIC_LIST_HEAD(LdrpDllNotificationList);
+
+/* Initialize critical section statically */
+
+static RTL_CRITICAL_SECTION LdrpDllNotificationLock;
+
+static RTL_CRITICAL_SECTION_DEBUG LdrpDllNotificationLockDebug = {
+    .CriticalSection = &LdrpDllNotificationLock
+};
+
+static RTL_CRITICAL_SECTION LdrpDllNotificationLock = {
```

Comment:
I am not a big fan of these static intialized critical sections, because they trigger application verifier.

We should either decide to embrace these, and add some compatibility thing for verifier, or not use them at all.

@tkreuzer thoughts on this?

Reply:
What does application verifier complain about?